app/src/main/java/com/odysee/app/ui/golive/GoLiveFragment.java

```diff
@@ -0,0 +1,324 @@
+            }
+        });
+        executor.shutdown();
+    }
+
+    private Claim buildLivestreamClaim() {
+        Claim claim = new Claim();
+
+        claim.setName("livestream-" + System.currentTimeMillis());
+        claim.setAmount("0.01");
+
+        Claim.StreamMetadata metadata = new Claim.StreamMetadata();
+        metadata.setTitle(Helper.getValue(inputTitle.getText()));
+
+        Claim selectedChannel = (Claim) selectChannelSpinner.getSelectedItem();
+        if (selectedChannel != null && !selectedChannel.isPlaceholder() && !selectedChannel.isPlaceholderAnonymous()) {
+            this.selectedChannel = selectedChannel;
+            claim.setSigningChannel(selectedChannel);
+        }
+
+        claim.setValueType(Claim.TYPE_STREAM);
+        claim.setValue(metadata);
+
+        return claim;
+    }
+
+    private void publishLivestreamClaim() {
+        Claim claim = buildLivestreamClaim();
+        AccountManager am = AccountManager.get(getContext());
+        String authToken = am.peekAuthToken(Helper.getOdyseeAccount(am.getAccounts()), "auth_token_type");
+        PublishClaimTask task = new PublishClaimTask(claim, "", null, authToken, new ClaimResultHandler() {
+            @Override
+            public void beforeStart() {
+
+            }
+
+            @Override
+            public void onSuccess(Claim claimResult) {
+                signAndSetupStream();
+            }
+
+            @Override
+            public void onError(Exception error) {
+                showError(error.getMessage());
+            }
+        });
+        task.executeOnExecutor(AsyncTask.THREAD_POOL_EXECUTOR);
+    }
+
+    private void updateChannelList(List<Claim> channels) {
+        selectChannelSpinner.setAdapter(new InlineChannelSpinnerAdapter(getContext(), R.layout.spinner_item_channel, channels));
+    }
+
+    private void fetchChannels() {
+        if (Lbry.ownChannels != null && Lbry.ownChannels.size() > 0) {
+            updateChannelList(Lbry.ownChannels);
+            return;
+        }
+
+        ClaimListTask task = new ClaimListTask(Claim.TYPE_CHANNEL, null, new ClaimListResultHandler() {
+            @Override
+            public void onSuccess(List<Claim> claims) {
+                Lbry.ownChannels = new ArrayList<>(claims);
+                updateChannelList(Lbry.ownChannels);
+            }
+
+            @Override
+            public void onError(Exception error) {
+                if (error != null) {
+                    showError(error.getMessage());
+                }
+            }
+        });
+        task.executeOnExecutor(AsyncTask.THREAD_POOL_EXECUTOR);
+    }
+
+    private String createStreamKey(String signature, String signingTs) {
+        String hexData = toHexString(selectedChannel.getName());
+        return selectedChannel.getClaimId()
+                + "?d=" + hexData
+                + "&s=" + signature
+                + "&t=" + signingTs;
+    }
+
+    private String toHexString(final String value) {
```

Comment:
This method will serve better if moved to Helper class. There are two other places where this same code is used and it is not depending on any Android framework variable or method. You could also move the other usages I mentioned to use the one from the Helper class.